whois-api/src/main/java/net/ripe/db/whois/api/fulltextsearch/ElasticFullTextRebuild.java

```diff
@@ -0,0 +1,263 @@
+import org.elasticsearch.client.indices.GetIndexResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Component;
+
+import javax.sql.DataSource;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import static net.ripe.db.whois.api.elasticsearch.ElasticSearchConfigurations.getMappings;
+import static net.ripe.db.whois.api.elasticsearch.ElasticSearchConfigurations.getSettings;
+
+@Component
+public class ElasticFullTextRebuild {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ElasticFullTextRebuild.class);
+
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyyMMdd-HHmmss");
+
+    private final ElasticIndexService elasticIndexService;
+
+    private static final int BATCH_SIZE = 100;
+    private final JdbcTemplate jdbcTemplate;
+    private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+    private final String source;
+    private final List<String> elasticHosts;
+
+    @Autowired
+    public ElasticFullTextRebuild(final ElasticIndexService elasticIndexService,
+                                  @Value("#{'${elastic.host:}'.split(',')}") final List<String> elasticHosts,
+                                  @Qualifier("whoisSlaveDataSource") final DataSource dataSource,
+                                  @Value("${whois.source}") final String source) {
+        this.elasticIndexService = elasticIndexService;
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+        this.source = source;
+        this.elasticHosts = elasticHosts;
+        this.namedParameterJdbcTemplate = new NamedParameterJdbcTemplate(jdbcTemplate);
+    }
+
+    public void run() throws IOException {
+        LOGGER.info("Source is  {}", source);
+        final Stopwatch stopwatch = Stopwatch.createStarted();
+
+        final String indexName = "whois-" + DATE_TIME_FORMATTER.format(LocalDateTime.now());
+
+        rebuild(indexName, elasticIndexService.getMetadataIndex(), true);
+
+        deleteOldIndexes(indexName);
+
+        LOGGER.info("ES indexing complete {}", stopwatch);
+    }
+
+    public void rebuild(final String indexName, final String metadataName, final boolean shouldSetAlias) throws IOException {
+
+        final RestHighLevelClient client = elasticIndexService.getClient();
```

Comment:
Surround with try-with-resources so the client is autoclosed ?

Reply:
Not really, the client from a spring component

Reply:
It's from ElasticSearch and is marked as Closeable so I think we should close it to be sure
https://artifacts.elastic.co/javadoc/org/elasticsearch/client/elasticsearch-rest-high-level-client/7.16.0/org/elasticsearch/client/RestHighLevelClient.html

Reply:
OK I see `getClient()` agree it should not be closed.